web/src/containers/Auth.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { ACCESS_CONTROL } from '../constants/auth';
+import { SpinLoader } from '../components/UI';
+import { AuthModal } from '../components/Auth/AuthModal';
+import { authorizationAction } from '../actionCreators';
+
+export class Auth extends Component {
+  static displayName = '[Component Auth]';
+  render() {
+    const {
+      isAuthenticated,
+      userInfo,
+      userId,
+      authorizationStarted,
+      authCode,
+      errorMessage,
+      authorizationAction,
+    } = this.props;
+
+    if (isAuthenticated && !authorizationStarted && !authCode && !errorMessage)
+      authorizationAction(userId, userInfo.roles);
+
+    let redirect = null;
+    if (authCode === ACCESS_CONTROL.NO_ROLE) redirect = <Redirect to="/notAuthorized" />;
+    else if (
+      authCode === ACCESS_CONTROL.REQUESTER_ROLE ||
+      authCode === ACCESS_CONTROL.REVIEWER_ROLE
+    )
+      redirect = <Redirect to="/home" />;
+
+    const content = isAuthenticated ? (
+      authorizationStarted ? (
```

Comment:
nested ternary's are difficult to read, i'd suggest expanding this to if/else ifs